File: tests/fixtures/safe/safe.java
// FIXTURE: the SAFE versions, Java.
// EXPECT-NONE
import java.security.MessageDigest;
import java.sql.*;

public class SafeDao {
    public ResultSet findById(Connection conn, String id) throws SQLException {
        PreparedStatement ps = conn.prepareStatement("SELECT * FROM users WHERE id = ?");
        ps.setString(1, id);
        return ps.executeQuery();
    }

    private static final String DB_PASSWORD = System.getenv("DB_PASSWORD");

    public MessageDigest strong() throws Exception {
        return MessageDigest.getInstance("SHA-256");
    }
}

File: tests/fixtures/safe/safe.ts
// FIXTURE: the SAFE versions, TypeScript.
// EXPECT-NONE
interface Conn { execute(sql: string, params: unknown[]): Promise<unknown>; }

export async function findOrder(conn: Conn, orderId: string): Promise<unknown> {
  return conn.execute("SELECT * FROM orders WHERE order_id = ?", [orderId]);
}

export const apiKey: string | undefined = process.env['API_KEY'];
